arrow_test.go

```diff
@@ -34,6 +36,56 @@ func TestCheckVersion(t *testing.T) {
 	println(s)
 }
 
+func TestArrowBatchHighPrecision(t *testing.T) {
+	runDBTest(t, func(dbt *DBTest) {
+		ctx := WithArrowBatches(context.Background())
+		query := "select '0.1':: DECIMAL(38, 19) as c"
+
+		var rows driver.Rows
+		var err error
+
+		// must use conn.Raw so we can get back driver rows (an interface)
+		// which can be cast to snowflakeRows which exposes GetArrowBatch
+		err = dbt.conn.Raw(func(x interface{}) error {
+			queryer, implementsQueryContext := x.(driver.QueryerContext)
+			assertTrueF(t, implementsQueryContext, "snowflake connection driver does not implement queryerContext")
+
+			rows, err = queryer.QueryContext(WithArrowBatches(ctx), query, nil)
+			return err
+		})
+
+		if err != nil {
+			t.Error(err, "error running select query")
+		}
+
+		sfRows, isSfRows := rows.(SnowflakeRows)
+		assertTrueF(t, isSfRows, "rows should be snowflakeRows")
+
+		arrowBatches, err := sfRows.GetArrowBatches()
+		if err != nil {
+			t.Error(err, "error getting arrow batches")
+		}
+
+		if len(arrowBatches) == 0 {
```

Comment:
If you really want to make gosnowflake better, I strongly encourage you to introduce `assertNotEqualF` :) But if you don't have time, we can leave it as is.

Reply:
Done!